src/posix/platform/system.cpp

```diff
@@ -129,52 +131,56 @@ void otSysSetInfraNetif(const char *aInfraNetifName, int aIcmp6Socket)
 
 void platformInit(otPlatformConfig *aPlatformConfig)
 {
-    CoprocessorType type;
-
 #if OPENTHREAD_POSIX_CONFIG_BACKTRACE_ENABLE
     platformBacktraceInit();
 #endif
 
     platformAlarmInit(aPlatformConfig->mSpeedUpFactor, aPlatformConfig->mRealTimeSignal);
 
-    type = platformSpinelManagerInit(get802154RadioUrl(aPlatformConfig));
-    if (type != OT_COPROCESSOR_RCP)
+    sCoprocessorType = platformSpinelManagerInit(get802154RadioUrl(aPlatformConfig));
+    if (sCoprocessorType == OT_COPROCESSOR_RCP)
+    {
+        platformRadioInit(get802154RadioUrl(aPlatformConfig));
+    }
+    else if (sCoprocessorType != OT_COPROCESSOR_NCP)
     {
-        printf("Only RCP is supported!\n");
+        printf("Unknown sCoprocessorType of the co-processor!\n");
         exit(OT_EXIT_FAILURE);
     }
+    aPlatformConfig->mCoprocessorType = sCoprocessorType;
 
-    platformRadioInit(get802154RadioUrl(aPlatformConfig));
-
-    // For Dry-Run option, only init the radio.
+    // For Dry-Run option, only init the co-processor.
     VerifyOrExit(!aPlatformConfig->mDryRun);
 
+    if (sCoprocessorType == OT_COPROCESSOR_RCP)
+    {
 #if OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
-    platformTrelInit(getTrelRadioUrl(aPlatformConfig));
+        platformTrelInit(getTrelRadioUrl(aPlatformConfig));
 #endif
-    platformRandomInit();
+        platformRandomInit();
 
 #if OPENTHREAD_POSIX_CONFIG_INFRA_IF_ENABLE
-    ot::Posix::InfraNetif::Get().Init();
+        ot::Posix::InfraNetif::Get().Init();
 #endif
 
 #if OPENTHREAD_CONFIG_MULTICAST_DNS_ENABLE
-    ot::Posix::MdnsSocket::Get().Init();
+        ot::Posix::MdnsSocket::Get().Init();
 #endif
 
-    gNetifName[0] = '\0';
+        gNetifName[0] = '\0';
 
 #if OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
-    platformNetifInit(aPlatformConfig);
+        platformNetifInit(aPlatformConfig);
 #endif
 
 #if OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
 #if OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
-    ot::Posix::Udp::Get().Init(otSysGetThreadNetifName());
+        ot::Posix::Udp::Get().Init(otSysGetThreadNetifName());
 #else
-    ot::Posix::Udp::Get().Init(aPlatformConfig->mInterfaceName);
+        ot::Posix::Udp::Get().Init(aPlatformConfig->mInterfaceName);
 #endif
 #endif
```

Comment:
The list of initialization could grow in the future which makes it harder to read. Should we consider wrapping it into another function?

Reply:
Make sense! I've encapsulated the processes in different functions.